File: ios/Classes/Entities/InitializationStatus.swift
import GoogleMobileAds

enum InitializationStatus : uint {
    
    init(status: GADInitializationStatus){
        let adapterStatuses = status.adapterStatusesByClassName
        let allAdaptersReady = adapterStatuses.allSatisfy { $0.value.state == .ready }
        
        self = allAdaptersReady ? .success : .fail
    }
    
    case success = 0
    case fail = 1
}

File: lib/src/entities/initialization_status.dart
/// Describes SDK initialization status
enum InitializationStatus {
  success(0),
  fail(1),
  fallbackPolling(2);

  const InitializationStatus(this.code);

  final int code;

  static InitializationStatus? fromCode(int code) {
    return switch (code) {
      0 => InitializationStatus.success,
      1 => InitializationStatus.fail,
      2 => InitializationStatus.fallbackPolling,
      _ => null,
    };
  }
}
